client/trino-client/src/main/java/io/trino/client/uri/TrinoUri.java

```diff
@@ -0,0 +1,1250 @@
+    {
+        return password.isPresent();
+    }
+
+    public Optional<String> getSessionUser()
+    {
+        return sessionUser;
+    }
+
+    public Map<String, ClientSelectedRole> getRoles()
+    {
+        return roles.orElse(ImmutableMap.of());
+    }
+
+    public Optional<String> getApplicationNamePrefix()
+    {
+        return applicationNamePrefix;
+    }
+
+    public Map<String, String> getExtraCredentials()
+    {
+        return extraCredentials.orElse(ImmutableMap.of());
+    }
+
+    public Optional<String> getClientInfo()
+    {
+        return clientInfo;
+    }
+
+    public Optional<String> getClientTags()
+    {
+        return clientTags;
+    }
+
+    public Optional<String> getTraceToken()
+    {
+        return traceToken;
+    }
+
+    public Map<String, String> getSessionProperties()
+    {
+        return sessionProperties.orElse(ImmutableMap.of());
+    }
+
+    public Optional<String> getSource()
+    {
+        return source;
+    }
+
+    public boolean isCompressionDisabled()
+    {
+        return disableCompression.orElse(false);
+    }
+
+    public boolean isAssumeLiteralNamesInMetadataCallsForNonConformingClients()
+    {
+        return assumeLiteralNamesInMetadataCallsForNonConformingClients.orElse(false);
+    }
+
+    public boolean isAssumeLiteralUnderscoreInMetadataCallsForNonConformingClients()
+    {
+        return assumeLiteralUnderscoreInMetadataCallsForNonConformingClients.orElse(false);
+    }
+
+    public Properties getProperties()
+    {
+        return properties;
+    }
+
+    public static DriverPropertyInfo[] getPropertyInfo(String url, Properties info)
+    {
+        Properties properties = urlProperties(url, info);
+
+        return ConnectionProperties.allProperties().stream()
+                .filter(property -> property.isValid(properties))
+                .map(property -> property.getDriverPropertyInfo(properties))
+                .toArray(DriverPropertyInfo[]::new);
+    }
+
+    private static Properties urlProperties(String url, Properties info)
+    {
+        try {
+            return create(url, info).getProperties();
+        }
+        catch (SQLException e) {
+            return info;
+        }
+    }
+
+    public Consumer<OkHttpClient.Builder> getSetupSsl()
+    {
+        if (!useSecureConnection) {
+            return OkHttpUtil::setupInsecureSsl;
+        }
+        SslVerificationMode sslVerificationMode = sslVerification.orElse(FULL);
+        if (sslVerificationMode.equals(NONE)) {
+            return OkHttpUtil::setupInsecureSsl;
+        }
+        return builder -> setupSsl(
+                builder,
+                sslKeyStorePath,
+                sslKeyStorePassword,
+                sslKeyStoreType,
+                sslTrustStorePath,
+                sslTrustStorePassword,
+                sslTrustStoreType,
+                sslUseSystemTrustStore.orElse(false));
+    }
+
+    public void setupClient(OkHttpClient.Builder builder)
```

Comment:
in both of the usages we are passing new, fresh OkHttpClient.Builder here.

Can we instead do: `public OkHttpClient.Builder toHttpClientBuilder()` ?